src/app/event/Modal.tsx

```diff
@@ -0,0 +1,117 @@
+import { useState } from "react";
+import { isAddress } from "viem";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { useFieldArray, useForm } from "react-hook-form";
+
+import {
+  AssignTicketText,
+  CloseDialogButton,
+  TriggerDialogButton,
+} from "./components/ModalComponents";
+
+interface FormData {
+  addresses: { address: string }[];
+}
+
+export function Modal() {
+  const [open, setOpen] = useState(false);
+  const {
+    register,
+    control,
+    handleSubmit,
+    watch,
+    formState: { errors, isValid },
+  } = useForm<FormData>({
+    defaultValues: {
+      addresses: [{ address: "" }],
+    },
+    mode: "onChange",
+  });
+
```

Comment:
I think a big improvement would be to send all this useForms logic to a separate hook, keeping it all outside the modal component and using it like this:

const { register, fields, errors, isValid, handleSubmit, handleNumberOfFields } = useAddressForm();

Reply:
Your hook will be something like that:
```
const useAddressForm = () => {
  const {
    register,
    control,
    handleSubmit,
    watch,
    formState: { errors, isValid },
  } = useForm<FormData>({
    defaultValues: {
      addresses: [{ address: "" }],
    },
    mode: "onChange",
  });

  const { fields, append, remove } = useFieldArray({
    control,
    name: "addresses",
  });

  const watchedFields = watch("addresses");

  const handleNumberOfFields = useCallback(() => {
    const lastElement = watchedFields[watchedFields.length - 1];
    if (isAddress(lastElement?.address)) {
      append({ address: "" });
    }
    const emptyElements = watchedFields
      .map((field, index) => (field.address === "" ? index : -1))
      .filter((index) => index !== -1);
    if (emptyElements.length > 1) {
      remove(emptyElements[0]);
    }
  }, [watchedFields, append, remove]);

  return {
    register,
    fields,
    errors,
    isValid,
    handleSubmit,
    handleNumberOfFields,
  };
};
```